feat(Other): Changes in preparation for Zephyr HPB support

petejohanson-adi opened the pull request:
### Description

The following two commits are prep work for the Zephyr HPB memory-controller driver. They are tested and confirmed working with that driver.

The `const` flags are a nicety, to avoid compiler warnings related to passing in our const structures that are generated from the device tree there.

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

ttmut commented on Libraries/zephyr/MAX/Include/wrap_max32_emcc.h:
```diff
@@ -0,0 +1,43 @@
+extern "C" {
+#endif
+
+static inline void Wrap_MXC_EMCC_Enable(void)
+{
+    MXC_EMCC_Enable();
+}
+
+static inline void Wrap_MXC_EMCC_Disable(void)
+{
+    MXC_EMCC_Disable();
+}
```
You can just call `MXC_EMCC_Enable()` from inside the Zephyr driver as long as the function prototype is the same for all parts. Wrapping is not needed in that case.

petejohanson-adi replied:
Makes sense, thanks! Removed that commit from the branch.

ttmut replied:
It looks like the same can be applied to HPB wrapper as well.

petejohanson-adi replied:
Indeed. Rebased and removed the commit, and updated my Zephyr driver work to make the MSDK call directly. Thanks.

petejohanson-adi commented:
@ttmut It *seems* that Ci does not like my PR to include more than one self-contained commit, based on my quick scan of the failure at https://github.com/analogdevicesinc/msdk/actions/runs/13792457914/job/38575624307?pr=1361

Would you rather I:

* Split into two PRs, OR
* Squash into one commit, OR
* Leave it.

TIA.

petejohanson-adi commented:
> 
> > Would you rather I:
> > 
> > * Split into two PRs, OR
> > * Squash into one commit, OR
> > * Leave it.
> > 
> > TIA.
> 
> Split into two PRs

Split into #1362 and #1363 and rebased both on latest `main`. CI seems to get past the "Get Changed Files" step, so closing this PR in favor of those two discrete PRs.
